File: lib/models/coordinate.dart
/// 三维坐标系统
/// X轴：随机变量（不可控因素）
/// Y轴：当前价值（核心竞争指标）
/// Z轴：影响力（长期积累值）
class Coordinate {
  final double x; // 随机变量
  final double y; // 当前价值
  final double z; // 影响力

  Coordinate({
    required this.x,
    required this.y,
    required this.z,
  });

  // 从JSON创建
  factory Coordinate.fromJson(Map<String, dynamic> json) {
    return Coordinate(
      x: (json['x'] as num).toDouble(),
      y: (json['y'] as num).toDouble(),
      z: (json['z'] as num).toDouble(),
    );
  }

  // 转换为JSON
  Map<String, dynamic> toJson() {
    return {
      'x': x,
      'y': y,
      'z': z,
    };
  }

  // 创建副本并更新某些值
  Coordinate copyWith({
    double? x,
    double? y,
    double? z,
  }) {
    return Coordinate(
      x: x ?? this.x,
      y: y ?? this.y,
      z: z ?? this.z,
    );
  }

  // 计算综合分数（用于排名）
  double get totalScore => y + z;

  @override
  String toString() {
    return 'Coordinate(x: ${x.toStringAsFixed(2)}, y: ${y.toStringAsFixed(2)}, z: ${z.toStringAsFixed(2)})';
  }
}

File: cocos-version/assets/scripts/models/Coordinate.ts
/**
 * 三维坐标系统
 * X轴：随机变量（不可控因素）
 * Y轴：当前价值（核心竞争指标）
 * Z轴：影响力（长期积累值）
 */
export class Coordinate {
    public x: number; // 随机变量 (0-100)
    public y: number; // 当前价值 (0-200)
    public z: number; // 影响力 (10+)

    constructor(x: number, y: number, z: number) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    /**
     * 从JSON创建坐标
     */
    static fromJson(json: any): Coordinate {
        return new Coordinate(
            json.x || 0,
            json.y || 0,
            json.z || 0
        );
    }

    /**
     * 转换为JSON
     */
    toJson(): any {
        return {
            x: this.x,
            y: this.y,
            z: this.z
        };
    }

    /**
     * 创建副本
     */
    copyWith(options?: { x?: number; y?: number; z?: number }): Coordinate {
        return new Coordinate(
            options?.x !== undefined ? options.x : this.x,
            options?.y !== undefined ? options.y : this.y,
            options?.z !== undefined ? options.z : this.z
        );
    }

    /**
     * 计算综合分数（用于排名）
     */
    get totalScore(): number {
        return this.y + this.z;
    }

    toString(): string {
        return `Coordinate(x: ${this.x.toFixed(2)}, y: ${this.y.toFixed(2)}, z: ${this.z.toFixed(2)})`;
    }
}
